### PR TITLE
replicated the changes made in ent repo for consistency

### PR DESCRIPTION
**Pull request by mallikabandaru**
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

**Review by huikang — APPROVED** [review]

File: test-integ/topoutil/asserter.go [hunk 1]
```diff
@@ -4,26 +4,26 @@
...
-
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/consul/proto-public/pbresource"
 	"github.com/hashicorp/consul/sdk/testutil"
 	"github.com/hashicorp/consul/sdk/testutil/retry"
 	libassert "github.com/hashicorp/consul/test/integration/consul-container/libs/assert"
 	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
 	"github.com/hashicorp/consul/testing/deployer/topology"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
```
Comment (huikang) [inline test-integ/topoutil/asserter.go:18]: nit: please format the imports

File: test-integ/topoutil/asserter.go [hunk 2]
```diff
@@ -439,3 +439,115 @@ func (a *Asserter) TokenExist(t *testing.T, cluster string, node *topology.Node,
...
+	boolIsErrorJSON := false
+	jsonData, err := os.ReadFile(filePath)
+	require.NoError(t, err)
+
+	// Print details of each AuditEntry
+	var entry AuditEntry
+	events := strings.Split(strings.TrimSpace(string(jsonData)), "\n")
+
+	// function to validate the error object when the health endpoint is returning 429
+	isErrorJSONObject := func(errorString string) bool {
```
Comment (jm96441n) [inline test-integ/topoutil/asserter.go:521]: why can't this just return an error instead of a bool?

File: test-integ/topoutil/asserter.go [hunk 3]
```diff
@@ -439,3 +439,115 @@ func (a *Asserter) TokenExist(t *testing.T, cluster string, node *topology.Node,
...
+		err := json.Unmarshal([]byte(errorString), &m)
+		if err != nil {
+			// Error occurred during unmarshaling, indicating the string is not a JSON object
+			return false
+		}
+		return true
+	}
+
+	for _, event := range events {
+		if strings.Contains(event, "/v1/operator/autopilot/health") {
```
Comment (jm96441n) [inline test-integ/topoutil/asserter.go:533]: can we flatten out these nested conditionals? like the first two can just be:
```
if (strings.Contains(event, "/v1/operator/autopilot/health") && strings.Contains(event, "OperationComplete")
```

File: test-integ/topoutil/asserter.go [hunk 4]
```diff
@@ -439,3 +439,115 @@ func (a *Asserter) TokenExist(t *testing.T, cluster string, node *topology.Node,
...
+			if strings.Contains(event, "OperationComplete") {
+				err = json.Unmarshal([]byte(event), &entry)
+				require.NoError(t, err, "Error unmarshalling JSON: %v\", err")
+				if entry.Payload.Response.Status == "429" {
+					fmt.Printf("Payload Response Error: %s\n", entry.Payload.Response.Error)
+					errResponse := entry.Payload.Response.Error
+					if isErrorJSONObject(errResponse) {
+						fmt.Println("Autopilot Health endpoint error response in the audit log looks fine")
```
Comment (jm96441n) [inline test-integ/topoutil/asserter.go:541]: are these debugging statements that were left in? we should be able to remove these then and get rid of the `else` branch here

File: test-integ/topoutil/asserter.go [hunk 5]
```diff
@@ -439,3 +439,115 @@ func (a *Asserter) TokenExist(t *testing.T, cluster string, node *topology.Node,
...
+						boolIsErrorJSON = true
+						break
+					} else {
+						fmt.Println("Autopilot Health endpoint error response in the audit log is in unexpected format")
+						break
+					}
+				}
+			}
+		}
+	}
```
Comment (jm96441n) [inline test-integ/topoutil/asserter.go:551]:
```suggestion
for _, event := range events {
		if !strings.Contains(event, "/v1/operator/autopilot/health") || !strings.Contains(event, "OperationComplete") {
		   continue
		}

		err = json.Unmarshal([]byte(event), &entry)
		require.NoError(t, err, "Error unmarshalling JSON: %v\", err")
		if entry.Payload.Response.Status != "429" {
		    continue
		 }

		errResponse := entry.Payload.Response.Error
		err := isErrorJSONObject(errResponse)
	   require.NoError(t, err,  "Unable to verify audit log health endpoint error message")
		break
		


```

File: test-integ/topoutil/asserter.go (hunk 3 above)
Comment (mallikabandaru) [reply test-integ/topoutil/asserter.go:533]: done

**Pull request merged**